test/scripts/traffic-flow-tests.sh

```diff
@@ -0,0 +1,167 @@
+        if ! command -v "${cmd}" &> /dev/null; then
+            error_exit "Dependency not met: ${cmd}"
+        fi
+    done
+}
+
+usage() {
+    echo "Usage: $(basename "$0") {setup|run}"
+}
+
+create_config_file_defaults() {
+    local file_name="${1:-config.yaml}"
+
+    cat <<EOT > "$file_name"
+tft:
+  - name: "Github Workflow Test"
+    namespace: "default"
+    test_cases: "TBD"
+    duration: "10"
+    connections:
+      - name: "Connection_1"
+        type: "iperf-tcp"
+        instances: 1
+        server:
+          - name: "ovn-worker"
+            persistent: "false"
+            sriov: "false"
+        client:
+          - name: "ovn-worker2"
+            sriov: "false"
+        plugins: []
+kubeconfig:
+kubeconfig_infra:
+EOT
+}
+
+process_test_results() {
+    local result_file="$1"
+    local unique_tests_script='.["tft-tests"][]["flow_test"].tft_metadata.test_case_id'
+    local failures_script='
+      .["tft-tests"][] |
+      if .["flow_test"].success then empty
+      else {test_id: .["flow_test"].tft_metadata.test_case_id, message: .["flow_test"].msg}
+      end
+    '
+
+    log "INFO" "Processing test results from: ${result_file}"
+
+    local unique_test_ids
+    unique_test_ids=$(jq -r "${unique_tests_script}" "${result_file}" | sort -u)
+
+    echo "Test Cases:"
+    echo "-----------------------------------------------------"
+    echo "${unique_test_ids}"
+
+    local failures
+    failures=$(jq -c "${failures_script}" "${result_file}")
+
+    if [ -z "${failures}" ]; then
+        echo "-----------------------------------------------------"
+        echo "All traffic flow tests succeeded."
+    else
+        echo "-----------------------------------------------------"
+        echo "Some traffic flow tests failed. Details:"
+        echo "${failures}" | jq .
+        exit 1
+    fi
+}
+
+setup() {
+    check_dependencies
+
+    "${OCI_BIN}" pull "${TFT_TEST_IMAGE}" || error_exit "Failed to pull the test image."
+    local KIND_CLUSTER_NAME
+    KIND_CLUSTER_NAME=$(kind get clusters)
+    kind load docker-image "${TFT_TEST_IMAGE}" --name "${KIND_CLUSTER_NAME}" || error_exit "Failed to load the test image."
+
+    if [ -d "${TRAFFIC_FLOW_TESTS_FULL_PATH}" ]; then
+        error_exit "Install folder already exists: ${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+    fi
+    mkdir -pv "${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+    cd "${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+
+    git init
+    git remote add origin "${TRAFFIC_FLOW_TESTS_REPO}"
+    git fetch --depth=1 origin "${TRAFFIC_FLOW_TESTS_COMMIT}"
+    git checkout "${TRAFFIC_FLOW_TESTS_COMMIT}"
+
+    python -m venv tft-venv
+    source tft-venv/bin/activate
+    pip install --upgrade pip
+    pip install -r requirements.txt
+
+    local CONFIG_FILE="config.yaml"
+    create_config_file_defaults "$CONFIG_FILE"
+
+    sed -i "s|^kubeconfig:.*|kubeconfig: \"$KUBECONFIG\"|" "$CONFIG_FILE"
+    sed -i "s|test_cases:.*|test_cases: \"$TRAFFIC_FLOW_TESTS\"|" "$CONFIG_FILE"
```

Comment:
Ok, this works. Or maybe it would be nice to use `envsubst` instead and template the file that way?

maybe also have `create_config_file_defaults` print the result to stdout, and pip the sed commands together...

Reply:
you are right. It's fine as is.